test/unit_test/automl_test.cc

```diff
@@ -98,10 +98,11 @@ bool weights_offset_test(vw& all, multi_ex&)
   feature_indexes.emplace_back(hash_to_index(all.weights, get_index_for_feature(all, "User", "user=Anna")));
 
   const size_t interaction_index = interaction_to_index(all.weights,
-      get_index_for_feature(all, "Action", "article=sports"), get_index_for_feature(all, "User", "user=Tom"));
+      get_index_for_feature(all, "Action", "article=sports"), get_index_for_feature(all, "Action", "article=sports"));
```

Comment:
let's do an interaction that is not with itself like this one

Reply:
ah I see below another interaction, ok maybe leave for now. For some reason CI is failing these tests, it should be deterministic across platforms.

Reply:
the valgrind runs are off by a tiny bit, I think

Reply:
my guess for windows would be something related with the ordering of ties inside priority queue is different per platform.

Reply:
The issue doesn't seem platform specific because all the tests (Linux / Valgrind, Windows) are producing the same weights, they are just different from my local environment. I'll try adding a custom comparator to the priority queue to see if that helps.